test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,394 @@
+
+                resetValues();
+                EventQueue.invokeAndWait(() -> {
+                    currentComponent.setVisible(false);
+                    frame.invalidate();
+                    frame.validate();
+                });
+
+                robot.waitForIdle();
+                if (componentHidden) {
+                    throw new RuntimeException(
+                        "FAIL: ComponentHidden triggered when setVisible(false) "
+                            + "called for a hidden "
+                            + components[i].getClass());
+                }
+
+                resetValues();
+                EventQueue.invokeAndWait(() -> {
+                    currentComponent.setVisible(true);
+                    frame.invalidate();
+                    frame.validate();
+                });
+
+                robot.waitForIdle();
+                if (!componentShown) {
+                    throw new RuntimeException(
+                        "FAIL: ComponentShown not triggered for "
+                            + components[i].getClass());
+                }
+
+                resetValues();
+                EventQueue.invokeAndWait(() -> {
+                    currentComponent.setVisible(true);
+                    frame.invalidate();
+                    frame.validate();
+                });
+
+                robot.waitForIdle();
+                if (componentShown) {
+                    throw new RuntimeException(
+                        "FAIL: ComponentShown triggered when setVisible(true) "
+                            + "called for a shown " + components[i].getClass());
+                }
+
+                resetValues();
+                EventQueue.invokeAndWait(() -> {
+                    currentComponent.setLocation(
+                        currentComponent.getLocation().x + 1,
+                        currentComponent.getLocation().y);
+                    frame.invalidate();
+                    frame.validate();
+                });
+
+                robot.waitForIdle();
+                if (!componentMoved) {
+                    throw new RuntimeException(
+                        "FAIL: ComponentMoved not triggered for "
+                            + components[i].getClass());
+                }
+
+                resetValues();
+                EventQueue.invokeAndWait(() -> {
+                    currentComponent.setSize(
+                        currentComponent.getSize().width + 1,
+                        currentComponent.getSize().height);
+                    frame.invalidate();
+                    frame.validate();
+                });
+
+                robot.waitForIdle();
+                if (!componentResized) {
+                    throw new RuntimeException(
+                        "FAIL: ComponentResized not triggered for "
+                            + components[i].getClass());
+                }
+
+                // Disable the components and do the same set of operations
```

Comment:
if you're doing the same operations, you should write a helper function and call that function instead